flow/scene_update_context.cc

```diff
@@ -108,51 +109,27 @@ void SceneUpdateContext::CreateFrame(
       rrect.radii(SkRRect::kLowerRight_Corner).x(),  // bottom_right_radius
       rrect.radii(SkRRect::kLowerLeft_Corner).x()    // bottom_left_radius
   );
-  scenic::ShapeNode shape_node(session_);
-  shape_node.SetShape(shape);
-  shape_node.SetTranslation(shape_bounds.width() * 0.5f + shape_bounds.left(),
-                            shape_bounds.height() * 0.5f + shape_bounds.top(),
-                            0.f);
-  // TODO(SCN-1274): AddPart() and SetClip() will be deleted.
-  entity_node->AddPart(shape_node);
+  shape_node->SetShape(shape);
+  shape_node->SetTranslationRH(
+      shape_bounds.width() * 0.5f + shape_bounds.left(),
+      shape_bounds.height() * 0.5f + shape_bounds.top(), 0.f);
 
   // Check whether the painted layers will be visible.
   if (paint_bounds.isEmpty() || !paint_bounds.intersects(shape_bounds))
     paint_layers.clear();
 
   // Check whether a solid color will suffice.
   if (paint_layers.empty()) {
-    SetShapeColor(shape_node, color);
+    SetShapeColor(*shape_node, color);
     return;
   }
 
   // Apply current metrics and transformation scale factors.
   const float scale_x = ScaleX();
   const float scale_y = ScaleY();
 
-  // If the painted area only covers a portion of the frame then we can
```

Comment:
I *think* this is unneeded now, but would like a second opinion in case I'm off or this is indicating a deeper problem with this approach.

I was seeing textures being clipped here when they shouldn't have been because everything is flattened now. It looks like it's related to children being smaller than their parents. For example if there was a title bar, and then text in the title bar, with this on I was seeing the entire title bar cropped to just the size of the text line instead of the full bar. It looks like basically `paint_bounds` is a tiny subset of the full `ShapeNode` in this case, and not because the Node is partially offscreen like I'm interpreting the comment to mean.

Thoughts? /cc @chinmaygarde

Reply:
I think the old code is just a performance optimization when the `paint_bounds` is a tiny subset. To me, it doesn't feel like the comment suggests that the Node is partially offscreen.

To answer your question, yes I think this is unneeded, and probably incorrect to do when you only have one shape node.

In the old world with many shape nodes, there's one big shape node that handles all the background, so you can create a tiny inner shape node of size `paint_bounds` for painting.

In the new old with only one shape node, you can't reduce its size to `paint_bounds` without losing your background. And I think that's what you observed: if the size is reduced, then your title bar would only have a text, but no title bar background.

@chinmaygarde : can you please confirm or correct the above?